README.md

```diff
@@ -33,70 +33,72 @@ We will define a `set` and fill it up with the tokens which current query should
 
 ```python
 # fill up global tokens in our `set`
 global_tokens = ["<something>", "....."]
 key_tokens.update(global_tokens)
 ```
 
-* Introduce some random tokens which will transfer information by transfering to other tokens which in turn can transfer to other tokens. This may reduce the cost of information travel from one token to other. This is why `random` attention attention is introduced.
+* Introduce some random tokens which will transfer information by transfering to other tokens which in turn can transfer to other tokens. This may reduce the cost of information travel from one token to other.
 
 ```python
 # Let's add random tokens to our `set`
 random_tokens = ["<something>", "....."]
 key_tokens.update(random_tokens)
 ```
 
-Now, we just need our token to attend this `set` & possibly it will represent all the tokens nicely. Similar thing we will do for all the queries & attention matrix can be completed. But remember whole point is to do it as efficiently (& fastly) as we can & this is when bigbird block sparse attention comes into picture.
+Now, we just need our token to attend this `set` & possibly it will represent all the tokens nicely. The same approach is used for all other tokens. 
+But remember, the idea here is to approximate `BERT`'s full attention as efficiently as possible. This is when `BigBird`'s block sparse attention comes into the picture.
 
-### Understanding with Graphs
+### Understanding Big Bird's attention with Graphs
 
-Let's try to understand the need of `global`, `sliding` & `random` attention using the graphs.
+Let's try to understand the need of `global`, `sliding` & `random` attention using graphs.
 
 <img src="assets/global.png" width=256 height=200> </img>
 <img src="assets/sliding.png" width=256 height=200> </img>
 <img src="assets/random.png" width=256 height=200> </img> <br>
-*Above figure shows `global`, `sliding` & `random` connections respectively in graph, where each node corresponds to token and each dotted-line represents attention score. If no connection is made between 2 tokens, then attention score is assumed to 0.*
 
-BigBird block sparse attention is simply combination of these 3 figures. While in normal attention, all 81 connections (note: total 9 nodes are present) would have been present in above figure. You can simply think of normal attention as all the tokens being global.
+*The above figure shows `global` (left), `sliding` (middle) & `random` (right) connections respectively as a graph. Each node corresponds to a token and each dotted-line represents attention score. If no connection is made between 2 tokens, then attention score is assumed to 0.*
```

Comment:
I like the graphs! However I think you have to say that you assume that the attention matrix is symmetric, *i.e.* If A attends to B , then B attends to A. Otherwise the graph is not fully correct since there would be two connections between each node. 

Also in such a fully connected graph we don't have 81 edges but 8 + 7 + 6 + ... + 1 = 36 edges